File: exercise-solutions/module-1/09_Introduction_Classes/exercise-final/dotnet/ShoppingCartExercise/Classes/ShoppingCart.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace TechElevator.Classes
{
    public class ShoppingCart
    {
        public int TotalNumberOfItems { get; private set; }
        public decimal TotalAmountOwed { get; private set; }

        /**
        * The average price of all items that have been added to the cart.  This should be equal to
        * the totalAmountOwed divided by the totalNumberOfItems.
        *
        * @return the average price of items added to the cart
        */
        public decimal GetAveragePricePerItem()
        {
            if (this.TotalNumberOfItems == 0)
            {
                return 0;
            }
            else
            {
                return this.TotalAmountOwed / this.TotalNumberOfItems;
            }
        }

        /**
        * Adds items to the cart.
        *
        * @param numberOfItems the number of items being added to the cart
        * @param pricePerItem the price per item being added to the cart
        */
        public void AddItems(int numberOfItems, decimal pricePerItem)
        {
            this.TotalNumberOfItems += numberOfItems;
            this.TotalAmountOwed += (pricePerItem * numberOfItems);
        }

        /**
        * Removes all items from this cart.
        */
        public void Empty()
        {
            this.TotalNumberOfItems = 0;
            this.TotalAmountOwed = 0;
        }
    }
}

File: antoniohurt-java-blue-material/module-1/09_Introduction_Classes/student-exercise/src/main/java/com/techelevator/shoppingcart/ShoppingCart.java
package com.techelevator.shoppingcart;

public class ShoppingCart {

	private int totalNumberOfItems;
	private double totalAmountOwed;

	private double averagePricePerItem;

	

	public int getTotalNumberOfItems() {
		return totalNumberOfItems;
	}

	public double getTotalAmountOwed() {
		return totalAmountOwed;
	}  

	public void empty() {
		totalAmountOwed = 0;
		totalNumberOfItems = 0;
	}

	public void addItems(int numberOfItems, double pricePerItem) {
		totalNumberOfItems += numberOfItems;
		totalAmountOwed += (pricePerItem * numberOfItems);
	}

	public double getAveragePricePerItem() {
		if (totalNumberOfItems == 0) {
			return 0.0;
		} else {
			return totalAmountOwed / totalNumberOfItems;

		}

	}

}
